### ISSUE TITLE
Signing key changed?

### ISSUE THREAD

**Issue by IzzySoft** [aritra-tech/Notify#36]
Your last APK is signed by a different key than the one before. Release notes do not say why. Updates are not possible when a key was used that doesn't match the installed version. May I ask what happened?

* hash of the previous signing key: `e73b826b3b2946f0e66c456d6c3007e236b82efa258b55dc865a3747d1271f4c`
* hash of the latest signing key: `6b998202ca7ca7bf6376a70749c1cae6e07a2f34915eae9aea4fcd03c3c34555`

**Comment by aritra-tech** [comment]
Hey @IzzySoft, I apologize for the inconvenience actually my previous version and the latest version are very different actually I totally changed the version code from `1.0` to `1.1.0` so because of that the issue is coming up. I suggest you to please `Uninstall` the previous version and download the latest version.

**Comment by IzzySoft** [comment]
First, thanks for your response!

Changing `versionCode` or `versionName` does not result in different signing keys. For security reasons, we need to make sure it's really you (the original developer) – and not someone having gained access to your repo and tampered with it, providing their own APK with "some modifications". Different signing key hashes mean different signing keys (and I've checked, they are different) – and they are especially designed for this: to make sure the APK is legit.

So no offense meant (and I would have made this a "confidential issue" if Github would have that option – or rather contacted you by mail if there was a contact address mentioned here) – but can I be sure it's really you? What made you change the signing keys? Can you provide some APK of the current version signed with the original one for proof (e.g. attach it here to the issue – no need to "publish" it as if it was really you, which I still assume but need to ascertain, you will have had a reason for it)?

Your apology fully accepted, please accept mine too: as maintainer of the F-Droid repo where [your app is listed](https://apt.izzysoft.de/packages/com.aritra.notify/) I have a responsibility towards those using said F-Droid repository.

PS: With your app listed at Play I wonder how the update got (or will get) accepted there when signed with a different key. Details on that part I'd really like to hear, if you're willing to share them with me (honest question, no sarcasm).

**Comment by IzzySoft** [comment]
Ugh, no comment? So shall I rather remove the app from my repo then, as I cannot be sure it is save? Or did you just forget to hit "send" on your last comment? I've pointed out the security implications, and why I cannot take such a change lightly without proper checks. So ignoring that is quite disappointing and rather destroys trust instead of establishing it.

I saw there was a new release published, again using the newer signing key. It will still be rejected by my updater and by the Android clients. I will now disable update check for Notify in my repo until this is clarified, and will have to remove the listing if that won't be possible.

Again, no offense meant – but this is a security issue and not just some cosmetic change, so it should be properly addressed. Thanks for your understanding!
